### PR TITLE
Build with hyprland-session.service and hyprland-systemd.desktop

### PR DESCRIPTION
**Pull request by izmyname**
#### Describe your PR, what does it fix/add?
Improved systemd support. Hyprland is built with hyprland-session.service and hyprland-systemd.desktop from https://github.com/hyprwm/Hyprland/pull/7409.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
First serious merge, so better to double-check.

#### Is it ready for merging, or does it need work?
hyprland-session.service works, tested with `[ "$(tty)" = "/dev/tty1" ] && exec systemctl --user start --wait hyprland-session.service` in `.zprofile`. hyprland-systemd.desktop should work too.

### CODE REVIEW

**Comment by leiserfg** [comment]
Sorry for my lack of knowledge, but, won't this start hyprland on login regardless of what triggered the login, like if you have another WM/DM)?

**Comment by izmyname** [comment]
> Sorry for my lack of knowledge, but, won't this start hyprland on login regardless of what triggered the login, like if you have another WM/DM)?
If you mean my .zprofile command - yes, but you don't need it if you use DM.  In this case, you can choose hyprland-systemd.desktop inside your DM.

The command her eis to show, how I launch Hyprland and how I tested hyprland-session.service.

**Comment by leiserfg** [comment]
No, I was talking about the service, it will have to be `enable`d for that to happen tho. I'm just used to nix enabling the services for me so no problem.

**Comment by izmyname** [comment]
This service cannot be enabled, normally, as it lacks `[Install] WantedBy= ` condition. Unfortunately, I don't know anything about nix. Better to ask @fufexan .

**Comment by leiserfg** [comment]
If it can't be enabled then it's fine by me and my comments can be skipped, my fear was more about autostarting hyprland by accident.

File: CMakeLists.txt [hunk 1]
```diff
@@ -357,6 +357,14 @@ install(
 install(FILES ${CMAKE_SOURCE_DIR}/example/hyprland.desktop
         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/wayland-sessions)
 
+# session file - systemd support
+install(FILES ${CMAKE_SOURCE_DIR}/example/hyprland-systemd.desktop
+        DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/wayland-sessions)
+
+# install systemd service
+install(FILES ${CMAKE_SOURCE_DIR}/example/hyprland-session.service
+       DESTINATION "lib/systemd/user")
```
Comment (fufexan) [inline CMakeLists.txt:366]:
```suggestion
       DESTINATION "${CMAKE_INSTALL_LIBDIR}/systemd/user")
```

**Pull request closed without merging**